add a blog post: Debug a provisioned WildFly server with the help of …

liweinan opened the pull request:
…maven-wildfly-plugin

liweinan commented:
@bstansberry @jamezp Could you please help to check if this blog post is proper to be posted here? Thanks!

bstansberry commented on _posts/2024-10-02-debug-with-maven-wildfly-plugin.adoc:
```diff
@@ -0,0 +1,198 @@
+* I need to override some modules inside the WildFly server.
+* I need to run the WildFly server in debug mode, so I can use IDE tools to remotely debug the code.
+
+With the usage of `maven-wildfly-plugin`, the above goals can be achieved in an automated way. Here is an example of the configuration to show the usage of the plugin to achieve the above goals:
```
s/usage of/use of the/g  (first time)

for the 2nd one either

s/usage/use

or

s/the usage of/how to use/g

bstansberry commented on _posts/2024-10-02-debug-with-maven-wildfly-plugin.adoc:
```diff
@@ -0,0 +1,198 @@
+
+* https://github.com/jberet/jberet-examples/blob/main/deployment/pom.xml[jberet-examples/deployment/pom.xml at main · jberet/jberet-examples]
+
+The `wildfly-maven-plugin` configuration in the above `pom.xml` is like this:
```
s/is/looks

bstansberry commented on _posts/2024-10-02-debug-with-maven-wildfly-plugin.adoc:
```diff
@@ -0,0 +1,198 @@
+</plugin>
+----
+
+In the above configuration, the `provision` goal is configured to run in the `generate-test-resources` phase. In this execution, the `feature-packs` section is used to override the feature packs to be used by the provisioned server, and the `channels` section is using the `jberet-channel-manifest` channel to override the `batch-jberet` module to be used in the provisioned server. Please note that ...
```
s/is used to override/specifies/g

There is no overriding here, as there is no default feature pack that would be used if this wasn't set.

bstansberry commented on _posts/2024-10-02-debug-with-maven-wildfly-plugin.adoc:
```diff
@@ -0,0 +1,198 @@
+    ...
+</properties>
+----
+
```
Here I think you need a section explaining what the `jberet-channel-manifest` channel is. People likely don't really know what a channel is, and they don't know that JBeret publishes one. And you need to note that the 3.0.0.Final version of that manifest specifies use of JBeret 3.0.0 while WildFly 33.0.2 used 2.2.1.

liweinan replied:
done

bstansberry commented on _posts/2024-10-02-debug-with-maven-wildfly-plugin.adoc:
```diff
@@ -0,0 +1,198 @@
+</properties>
+----
+
+In the above way, I can define the WildFly version to be used and the JBeret version to be used in the provisioned WildFly server. To run this provisioned WildFly server in debug mode, the `maven-wildfly-plugin` provide options to support the debug mode. To check the usages of `wildfly-maven-plugin`, I can run the following command in the above example project to do so:
```
The first one should be 'wildfly-maven-plugin'

bstansberry commented on _posts/2024-10-02-debug-with-maven-wildfly-plugin.adoc:
```diff
@@ -0,0 +1,198 @@
+$ mvn help:describe -Dplugin=org.wildfly.plugins:wildfly-maven-plugin -Ddetail=true
+----
+
+The above command will output the full document of the `wildfly-maven-plugin`. And there are the relative parts of the document:
```
s/And there are/Following are/g

bstansberry commented on _posts/2024-10-02-debug-with-maven-wildfly-plugin.adoc:
```diff
@@ -0,0 +1,198 @@
+      attached.
+----
+
+As the document written in above, both the `wildfly:dev` and the `wildfly:run` goals support the `debug` mode, and the property to activate it is the `debug` option.
```
s/the `debug` option/the `wildfly.debug` option/g

bstansberry commented on _posts/2024-10-02-debug-with-maven-wildfly-plugin.adoc:
```diff
@@ -0,0 +1,198 @@
+[INFO] JAVA_OPTS : -Xms64m -Xmx512m -Djava.net.preferIPv4Stack=true -Djava.awt.headless=true -Djboss.modules.system.pkgs=org.jboss.byteman -agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=*:8787 --add-exports=java.base/sun.nio.ch=ALL-UNNAMED --add-exports=jdk.unsupported/sun.reflect=ALL-UNNAMED --add-exports=jdk.unsupported/sun.misc=ALL-UNNAMED --add-modules=java.se
+....
+
+As the output shown above, the debug options are added and the remote debug port is set as `8787` by default. This means the WildFly server is ready to accept the remote debug requests. In addition, the customized version of WildFly server codebase and the overridden version of the JBeret module codebase can be used for debugging now. I won’t introduce the way to use IDE to debug the WildFly serv ...
```
s/shown/shows
s/version of WildFly/version of the WildFly/g
s/to use IDE/to use an IDE/g
s/in before/before/g

bstansberry reviewed (changes requested):
@liweinan Very nice!

I made various copy editing suggestions. My only significant feedback is the need to explain a bit about the JBeret channel.

liweinan commented:
Thanks for the detailed review Brian! 

I'll update the text accordingly.

liweinan commented:
@bstansberry I have updated the text according to your advice, and I have added a section to explain the `jberet-channel-manifest`. Could you please help to review it again? IThanks!

liweinan commented:
I need to add some more contents.

liweinan commented:
done: finished adding the content related with manifest.

bstansberry commented on _posts/2024-10-02-debug-with-maven-wildfly-plugin.adoc:
```diff
@@ -0,0 +1,248 @@
+
+....
+[INFO] JAVA_OPTS : -Xms64m -Xmx512m -Djava.net.preferIPv4Stack=true -Djava.awt.headless=true -Djboss.modules.system.pkgs=org.jboss.byteman -agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=*:8787 --add-exports=java.base/sun.nio.ch=ALL-UNNAMED --add-exports=jdk.unsupported/sun.reflect=ALL-UNNAMED --add-exports=jdk.unsupported/sun.misc=ALL-UNNAMED --add-modules=java.se
+....
```
This needs to be ---- and not ....

With the .... this renders as one very long line, and not in a yellow block with a horizontal scroll bar on the bottom. The one line is twice as wide as the rest of the page.

liweinan replied:
done

bstansberry reviewed (changes requested):
Thanks @liweinan for the updates. There's one more rendering issue.

liweinan commented:
@bstansberry Yes sir!

jamezp commented on _posts/2024-10-02-debug-with-maven-wildfly-plugin.adoc:
```diff
@@ -0,0 +1,248 @@
+    <artifactId>wildfly-maven-plugin</artifactId>
+    <version>${version.maven.wildfly.plugin}</version>
+    <configuration>
+        <galleon-options>
+            <jboss-fork-embedded>true</jboss-fork-embedded>
+        </galleon-options>
```
I would remove this. I'll explain more below.

jamezp commented on _posts/2024-10-02-debug-with-maven-wildfly-plugin.adoc:
```diff
@@ -0,0 +1,248 @@
+                <goal>provision</goal>
+            </goals>
+            <configuration>
+                <provisioning-dir>${jboss.home}</provisioning-dir>
+                <galleon-options>
+                    <jboss-fork-embedded>${galleon.fork.embedded}</jboss-fork-embedded>
+                </galleon-options>
+                <feature-packs>
+                    <feature-pack>
+                        <groupId>${server.test.feature.pack.groupId}</groupId>
+                        <artifactId>${server.test.feature.pack.artifactId}</artifactId>
+                        <version>${wildfly.version}</version>
+                        <inherit-configs>false</inherit-configs>
+                        <included-configs>
+                            <config>
+                                <model>standalone</model>
+                                <name>standalone-full.xml</name>
+                            </config>
+                            <config>
+                                <model>standalone</model>
+                                <name>standalone.xml</name>
+                            </config>
+                        </included-configs>
+                        <excluded-packages>
+                            <name>docs.schema</name>
+                            <name>appclient</name>
+                            <name>domain</name>
+                        </excluded-packages>
+                    </feature-pack>
+                </feature-packs>
+                <channels>
+                    <channel>
+                        <manifest>
+                            <groupId>org.jberet</groupId>
+                            <artifactId>jberet-channel-manifest</artifactId>
+                            <version>${version.jberet}</version>
+                        </manifest>
+                    </channel>
+                </channels>
```
This configuration should all be moved above. The reason being if it's in the default configuration, it can be used from other goals specified on the command. For example `wildfly:dev`.

liweinan replied:
Thanks for the suggestion James! I have created a PR to move the configuration in the project: https://github.com/jberet/jberet-examples/pull/6

jamezp commented on _posts/2024-10-02-debug-with-maven-wildfly-plugin.adoc:
```diff
@@ -0,0 +1,248 @@
+modules/system/layers/base/org/jberet/jberet-core/main/jberet-core-3.0.0.Final.jar
+----
+
+To run the provisioned WildFly server in debug mode, the `wildfly-maven-plugin` provide options to support the debug mode. To check the usages of `wildfly-maven-plugin`, I can run the following command in the above example project to do so:
```
You could also link to the documentation where https://docs.wildfly.org/wildfly-maven-plugin.

liweinan replied:
Okay! I have added a `Note` section in below to provide the additional information.

jamezp reviewed (commented):
Just a couple minor comments, but otherwise looks good.

jamezp commented on _posts/2024-10-02-debug-with-maven-wildfly-plugin.adoc:
```diff
@@ -0,0 +1,248 @@
+
+[source,bash]
+----
+$ mvn help:describe -Dplugin=org.wildfly.plugins:wildfly-maven-plugin -Ddetail=true
```
You can also use 
```bash
mvn wildfly:help -Ddetail=true
```

Or 
```bash
mvn wildfly:help -Ddetail=true -Dgoal=dev
```

liweinan replied:
Thanks for teaching James!

bstansberry commented on _posts/2024-10-02-debug-with-maven-wildfly-plugin.adoc:
```diff
@@ -0,0 +1,248 @@
+
+And from the server output I can see the debug options are added:
+
+----
```
Please add a line before this with:

[source,txt]

It turns out changing the '....' to '----' wasn't necessary (but it is fine, so no need to change back). To get the rendering with the horizontal scroll bar it's the '[source...]' that's needed.

liweinan replied:
Yes sir!

liweinan commented:
@jamezp @bstansberry Thanks for the detailed comments! I'll modify the text accordingly.

liweinan commented:
@bstansberry @jamezp I have updated the text according to your suggestions. Please let me know if anything else need to be changed :D

bstansberry reviewed (approved):
@jamezp This LGTM, and I've verified it renders correctly. Please ping me in zulip when you've reviewed it and I'll do the GitHub action thing to move it to today's date.
